### DISCUSSION TITLE
ProMicro が正常に動作しない

### DISCUSSION THREAD

**Discussion by to4kwmr** [Yowkees/keyball#312] [Q&A]
ProMicro にファームウェア（Keyball61_via）の書き込みまで成功し、キースイッチのパッドをショートさせたもののキー入力されません。またOLEDにも何も表示されません。Windows PC に接続すると、数秒後にデバイス認識のエラーが出ます。
![error](https://github.com/Yowkees/keyball/assets/110336608/d29343f7-9f74-423b-9938-5fe05d341755)

なお、2つのProMicro のうち 1つは正常に動作しています。

・原因切り分けのため左右の PCB に取り付ける ProMicro を入れ替えて確認
-> 正常な ProMicro は PCB および OLED を入れ替えても動作するが、被疑の ProMicro は PCB および OLED を入れ替えても動作しない
　このことからソケットを含めた PCB および OLED は問題なく、ProMicro を疑っています。

・はんだ確認
-> 写真のピントがあわず若干見づらいですが、特にはんだ過不足などは無いように見えます　
![proMicro](https://github.com/Yowkees/keyball/assets/110336608/9178043d-b2a6-4720-891e-97e4b6c44ba9)

ProMicro の異常を疑い新たに ProMicro を購入することも考えましたが、12ピンヘッダが KeyBall に添付されるものと同じ高さのものが手に入らず、どうしたら良いものか困っています。。
何か切り分けや対処策などありましたらアドバイスいただけますと幸いです。

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

半田付けには問題ないように見えますので、ご指摘通りProMicroの不良と思われます。
12PピンヘッダーについてはKeyballシリーズご購入者様限定で部品販売可能です。
販売方法等ご相談したいので、下記白銀ラボウェブショップのお問い合わせよりご連絡いただければ幸いです。
https://shirogane-lab.net/inquiry

以上、お手数ですがよろしくお願いいたします。

**Reply by to4kwmr** [reply]
返信ありがとうございます！
白銀ラボウェブショップにて、お問い合わせさせていただきました。
よろしくお願いします。
